File: php/response/sms/DeliveryInfoNotification.php
<?php 

class DeliveryInfoNotification {

	protected $deliveryInfo=null;
	protected $callbackData=null;
	
	public function getDeliveryInfo() { return $this->deliveryInfo; }
	public function getCallbackData() { return $this->callbackData; }
	
	public function setDeliveryInfo($deliveryInfo) { $this->deliveryInfo=$deliveryInfo; }
	public function setCallbackData($callbackData) { $this->callbackData=$callbackData; }

	public function __construct($jsondata=null) {
		if (!is_null($jsondata)) {
			if (property_exists($jsondata, 'deliveryInfo')) {
				$deliveryInfoArray = $jsondata->deliveryInfo;
				if (is_array($deliveryInfoArray) && count($deliveryInfoArray)>0) {
					$processedDeliveryInfoArray = array();
					for ($i=0; $i<count($deliveryInfoArray); $i++) {
						$deliveryInfoElement=$deliveryInfoArray[$i];
						$processedDeliveryInfo=new DeliveryInfo($deliveryInfoElement);
						$processedDeliveryInfoArray[$i] = $processedDeliveryInfo;
					}
					$this->deliveryInfo = $processedDeliveryInfoArray;
				}
			}
			if (property_exists($jsondata, 'callbackData')) {
				$this->callbackData=$jsondata->callbackData;
			}
		}
	}

	public function toString() {
		$buffer='deliveryInfo=';
		if (!is_null($this->deliveryInfo) && is_array($this->deliveryInfo)) {
			$buffer=$buffer.'{';
			for ($i=0; $i<count($this->deliveryInfo); $i++) {
				if ($i>0) {
					$buffer=$buffer.', ['.$i.'] = '.$this->deliveryInfo[$i]->toString();
				} else {
					$buffer=$buffer.'['.$i.'] = '.$this->deliveryInfo[$i]->toString();
				}			
			}
			$buffer=$buffer.'}';
		}
		$buffer=$buffer.', callbackData='.$this->callbackData;
		return $buffer;
	}
		
}

?>	

File: Ruby/response/sms/DeliveryInfoNotification.rb
require "response/sms/DeliveryInfo"

class DeliveryInfoNotification

	def initialize
		@callbackData=nil
		@deliveryInfo=nil
	end

	def initializeJSON(jsondict)
		@callbackData=nil
		if (jsondict!=nil) && (jsondict.has_key?'callbackData') && (jsondict['callbackData']!=nil)
			@callbackData=jsondict['callbackData']
		end
		@deliveryInfo=nil
		if (jsondict!=nil) && (jsondict.has_key?'deliveryInfo') && (jsondict['deliveryInfo']!=nil) then
			@deliveryInfo=DeliveryInfo.new
			@deliveryInfo.initializeJSON(jsondict['deliveryInfo'])
		end
	end

	def getCallbackData
		@callbackData
	end

	def setCallbackData(callbackData)
		@callbackData=callbackData
	end

	def getDeliveryInfo
		@deliveryInfo
	end

	def setDeliveryInfo(deliveryInfo)
		@deliveryInfo=deliveryInfo
	end


end 
